src/quill.imageDrop.ts

```diff
@@ -72,15 +72,27 @@ export class ImageDrop {
   }
 
   private async handlePaste(evt: ClipboardEvent) {
-    await this.handleDataTransferList(evt.clipboardData?.items, evt);
+    if (evt.clipboardData) {
+      await this.handleDataTransferList(evt.clipboardData!, evt);
+    }
   }
 
-  private async handleDataTransferList(dataTransferItems: DataTransferItemList | undefined, evt: { preventDefault: () => void }) {
-    const items = Array.from(dataTransferItems || []);
+  private async handleDataTransferList(dataTransfer: DataTransfer, evt: Event) {
+    const items = Array.from(dataTransfer.items || []);
+
+    const html = dataTransfer.getData('text/html');
+    if (html) {
+      this.processHtml(html)
+        .catch(e => this.logger.error("error while processing pasted html", e));
+      evt.preventDefault();
+      return;
+    }
```

Comment:
I like this, but would rather it be behind an option that people can opt-in/out of:

``` tsx
 if (html && options.allowPasteHTML)
```

Reason being, the default behavior of this PR calls `dangerouslyPasteHTML` which probably shouldn't be on by default due to [XSS security concerns](https://quilljs.com/docs/modules/clipboard/#dangerouslypastehtml)

Reply:
Used the name `compressImagesInPastedHtml` instead, to be more clear about what it does.